### PR TITLE
fix: job cli returns different info from webui

### PR DESCRIPTION
**Pull request by eecsliu**
## Description
Changes the output of `det job list` to match what the webui displays.

## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: harness/determined/cli/job.py [hunk 1]
```diff
@@ -18,39 +19,37 @@ def ls(args: Namespace) -> None:
...
+            "#",
             "ID",
-            "Entity ID",
-            "Status",
             "Type",
-            "Slots Acquired",
-            "Slots Requested",
-            "Name",
+            "Job Name",
+            "Priority",
```
Comment (hamidzr) [inline harness/determined/cli/job.py:34]: blocking: this should cover fairshare with weights as well

**Review by hamidzr — CHANGES_REQUESTED** [review]
other than the issue I tagged it looks good

**Comment by hamidzr** [comment]
python 3.6 doesn't seem to have `fromisoformat` I think we do still advertise py3.6 support

https://docs.python.org/3.6/library/datetime.html
AttributeError: type object 'datetime.datetime' has no attribute 'fromisoformat'

**Review by hamidzr — COMMENTED** [review]
lgtm minus the python 3.6 support comment I added above.

image

nit: the original ask was to make it look like webui columns but I wonder whether the slots column is better of now compared to when it was reported as two columns. in the webui we get away with this since we can add on hover tool tips to explain what the numbers are. cc @shiyuann

File: harness/determined/cli/job.py [hunk 2]
```diff
@@ -1,56 +1,64 @@
...
 from determined.common.api.b import get_GetJobs
 from determined.common.declarative_argparse import Arg, Cmd, Group
 
 
 @authentication.required
 def ls(args: Namespace) -> None:
+    priority = True
+    config = api.get(args.master, "config").json()
+    try:
+        if config["resource_manager"]["scheduler"]["type"] == "fair_share":
```
Comment (hamidzr) [inline harness/determined/cli/job.py:21]: what if the resource pool config overrides it?

Comment (eecsliu) [reply harness/determined/cli/job.py:21]: ah good point. Will fix

**Comment by eecsliu** [comment]
For slots, what if the table read `Slots (acquired/needed)` ?

File: harness/determined/cli/job.py [hunk 3]
```diff
@@ -1,56 +1,75 @@
...
 from determined.cli.session import setup_session
+from determined.common import api
 from determined.common.api import authentication
 from determined.common.api.b import get_GetJobs
 from determined.common.declarative_argparse import Arg, Cmd, Group
 
 
 @authentication.required
 def ls(args: Namespace) -> None:
+    priority = True
```
Comment (hamidzr) [inline harness/determined/cli/job.py:19]: nit: would be nice to be able to tell the var type from the name.

File: harness/determined/cli/job.py [hunk 4]
```diff
@@ -1,56 +1,75 @@
...
+    config = api.get(args.master, "config").json()
+    try:
```
Comment (hamidzr) [inline harness/determined/cli/job.py:21]: suggestion: it seems safer and perhaps easier to me to go through the resource pool endpoint (the cli wouldn't need to assume this nested structure schema for the config.)

**Review by hamidzr — APPROVED** [review]
looks good. did you want to update the slots header? 
image (3)


following the feedback we got I updated the id column from the webui and tacked on the experiment id onto the name col. https://github.com/determined-ai/determined/pull/3266

I'm not sure if we want this for the cli but we can do that separately if we do.

**Pull request merged**